raiden-ts/src/raiden.ts

```diff
@@ -914,61 +912,34 @@ export class Raiden {
       this.log.info,
     );
 
-    const pathFindMeta = { tokenNetwork, target, value: decodedValue };
-    return merge(
-      // wait for pathFind response
-      this.action$.pipe(
-        first(isResponseOf(pathFind, pathFindMeta)),
+    const promise = this.action$
+      .pipe(
+        filter(isActionOf([transferSigned, transfer.failure])),
+        first(({ meta }) => meta.direction === Direction.SENT && meta.secrethash === secrethash),
         map((action) => {
-          if (pathFind.failure.is(action)) throw action.payload;
-          return action.payload.paths;
+          if (transfer.failure.is(action)) throw action.payload;
+          return transferKey(action.meta);
         }),
-      ),
-      // request pathFind; even if paths were provided, send it again for validation
-      // this is done at 'merge' subscription time (i.e. when above action filter is subscribed)
-      defer(() => {
-        this.store.dispatch(pathFind.request({ paths, pfs }, pathFindMeta));
-        return EMPTY;
-      }),
-    )
-      .pipe(
-        mergeMap((paths) =>
-          merge(
-            // wait for transfer response
-            this.action$.pipe(
-              filter(isActionOf([transferSigned, transfer.failure])),
-              first(
-                (action) =>
-                  action.meta.direction === Direction.SENT &&
-                  action.meta.secrethash === secrethash,
-              ),
-              map((action) => {
-                if (transfer.failure.is(action)) throw action.payload;
-                return transferKey(action.meta);
-              }),
-            ),
-            // request transfer with returned/validated paths at 'merge' subscription time
-            defer(() => {
-              this.store.dispatch(
-                transfer.request(
-                  {
-                    tokenNetwork,
-                    target,
-                    value: decodedValue,
-                    paymentId,
-                    secret,
-                    expiration,
-                    ...metadataFromPaths(paths),
-                  },
-                  { secrethash, direction: Direction.SENT },
-                ),
-              );
-              return EMPTY;
-            }),
-          ),
-        ),
       )
       .toPromise();
+    this.store.dispatch(
+      transfer.request(
```

Comment:
Wouldn't it be nicer to have a new action for this, instead of adding this `resolved` parameter? Something like `resolveTransfer`.

Reply:
I thought about it, but decided for the `resolved` tagged union because the whole `metadata` and most of the `payload` would overlap, and I'm not sure it's worth it to create a mostly similar copy of this action in order to split those. But I don't have hard feelings for it, it was mostly about code deduplication, but if you think it'd be better to have the copy, I can do it.